rector.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+use Rector\Config\RectorConfig;
+use Rector\Renaming\Rector\ClassConstFetch\RenameClassConstFetchRector;
+use Rector\Renaming\ValueObject\RenameClassConstFetch;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->paths([
+        __DIR__ . '/examples',
+        __DIR__ . '/src',
+        __DIR__ . '/tests',
+        __DIR__ . '/tools',
+        __FILE__,
+    ]);
+
+    // ReadPreference::RP_* constants are deprecated in favor of the string constants
+    $rectorConfig->ruleWithConfiguration(RenameClassConstFetchRector::class, [
+        new RenameClassConstFetch('MongoDB\Driver\ReadPreference', 'RP_PRIMARY', 'PRIMARY'),
```

Comment:
I haven't used rector before, but are class names typically string literals or should you prefer something like `MongoDB\Driver\ReadPreference::class`? I assume what you have here may be preferable to avoid any dependency on the extension itself.

Reply:
Code updated. That's better for code analysis, and the C extension is not required: `::class` constants are resolved at compile time.